packages/callbacks/tizen/inc/macros.h

```diff
@@ -6,379 +6,430 @@
 #define GET_CALLBACK_ID(CB_NAME, OFFSET) \
   __reserved_cb_id_array[BASE_CALLBACK_ID_##CB_NAME + OFFSET]
 
+#if __WORDSIZE == 64
+#define INVALID_USER_DATA(user_data) \
+  !user_data || reinterpret_cast<uintptr_t>(user_data) >> 32
+#define GET_CALLBACK_ID_FROM_USER_DATA(user_data) \
+  (uint32_t) reinterpret_cast<uintptr_t>(user_data)
+#else
+#define INVALID_USER_DATA(user_data) !user_data
+#define GET_CALLBACK_ID_FROM_USER_DATA(user_data) \
+  reinterpret_cast<uint32_t>(user_data)
+#endif
```

Comment:
Using macros [should be avoided](https://google.github.io/styleguide/cppguide.html#Preprocessor_Macros) as much as possible. Should the implementations differ in the two cases?

Reply:
Well, it does not need to. It could use the same expressions. Part of which would be a no-op on 32bit. Compiler was complaining about bit shift longer than operand but maybe it could be expresses differently.

Reply:
I just didn't understand why the bit shift operation would be necessary. Anyway if you're going to use these macros, please note that you can't use C-style casts: https://google.github.io/styleguide/cppguide.html#Casting.